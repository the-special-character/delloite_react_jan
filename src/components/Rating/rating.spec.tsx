import React from 'react';
import Rating from '.';
import { render, screen } from '@testing-library/react';

describe('Rating Component', () => {
  let container;
  beforeEach(() => {
    container = render(<Rating rate={4} count={120} />).container;
  });

  it('should render rating component', () => {
    if (container) {
      expect(container.firstChild).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    }
  });

  it('should count number of starts', () => {
    const startIcons = screen.queryAllByTestId('rating-svg');
    expect(startIcons.length).toBe(5);
  });

  it('first 4 starts should be dark gray and last light gray', () => {
    const startIcons = screen.queryAllByTestId('rating-svg');
    for (let i = 0; i < startIcons.length - 1; i++) {
      const element = startIcons[i];
      expect(element).toHaveClass('text-gray-800');
    }
    expect(startIcons[4]).toHaveClass('text-gray-200');
  });

  it('display count', () => {
    const link = screen.queryByRole('link');
    expect(link?.innerHTML).toMatch('120 reviews');
  });
});
