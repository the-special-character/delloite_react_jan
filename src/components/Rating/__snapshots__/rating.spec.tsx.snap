// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component should render rating component 1`] = `
<div
  class="mt-6"
>
  <h4
    class="sr-only"
  >
    Reviews
  </h4>
  <div
    class="flex items-center"
  >
    <div
      class="flex items-center"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5 flex-shrink-0 text-gray-800"
        data-testid="rating-svg"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-5 w-5 flex-shrink-0 text-gray-800"
        data-testid="rating-svg"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-5 w-5 flex-shrink-0 text-gray-800"
        data-testid="rating-svg"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-5 w-5 flex-shrink-0 text-gray-800"
        data-testid="rating-svg"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-5 w-5 flex-shrink-0 text-gray-200"
        data-testid="rating-svg"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.868 2.884c-.321-.772-1.415-.772-1.736 0l-1.83 4.401-4.753.381c-.833.067-1.171 1.107-.536 1.651l3.62 3.102-1.106 4.637c-.194.813.691 1.456 1.405 1.02L10 15.591l4.069 2.485c.713.436 1.598-.207 1.404-1.02l-1.106-4.637 3.62-3.102c.635-.544.297-1.584-.536-1.65l-4.752-.382-1.831-4.401z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="sr-only"
    >
      4
       out of 5 stars
    </p>
    <a
      class="ml-3 text-sm font-medium text-indigo-600 hover:text-indigo-500"
      href="#"
    >
      120
       reviews
    </a>
  </div>
</div>
`;
